src/dependency-registry.ts

```diff
@@ -0,0 +1,388 @@
+    );
+
+    const additionalCDN = DependencyAdditionalCDN.load(id);
+    if (!additionalCDN) {
+      return;
+    }
+    store.remove("DependencyAdditionalCDN", additionalCDN.id);
+
+    // Load each additional CDN after the removed one and decrement their index
+    for (
+      let i = event.params._additionalCDNIndex.toI32() + 1;
+      i < dependency.additionalCDNCount.toI32();
+      i++
+    ) {
+      const currentId = generateDependencyAdditionalCDNId(
+        dependency.id,
+        BigInt.fromI32(i)
+      );
+      const currentAdditionalCDN = DependencyAdditionalCDN.load(currentId);
+      if (currentAdditionalCDN) {
+        store.remove("DependencyAdditionalCDN", currentAdditionalCDN.id);
+        const updatedIndex = currentAdditionalCDN.index.minus(
+          BigInt.fromI32(1)
+        );
+        const updatedId = generateDependencyAdditionalCDNId(
+          dependency.id,
+          updatedIndex
+        );
+
+        const updatedAdditionalCDN = new DependencyAdditionalCDN(updatedId);
+        updatedAdditionalCDN.index = updatedIndex;
+        updatedAdditionalCDN.dependency = dependency.id;
+        updatedAdditionalCDN.cdn = currentAdditionalCDN.cdn;
+        additionalCDN.save();
+      }
+    }
+  }
+}
+
+export function handleDependencyAdditionalRepositoryUpdated(
+  event: DependencyAdditionalRepositoryUpdated
+): void {
+  const dependency = Dependency.load(event.params._dependencyType.toString());
+  if (!dependency) {
+    return;
+  }
+  const id = generateDependencyAdditionalRepositoryId(
+    dependency.id,
+    event.params._additionalRepositoryIndex
+  );
+
+  let additionalRepository = DependencyAdditionalRepository.load(id);
+
+  if (!additionalRepository) {
+    additionalRepository = new DependencyAdditionalRepository(id);
+    dependency.additionalRepositoryCount = dependency.additionalRepositoryCount.plus(
+      BigInt.fromI32(1)
+    );
+    dependency.save();
+  }
+
+  additionalRepository.dependency = dependency.id;
+  additionalRepository.repository = event.params._additionalRepository;
+  additionalRepository.index = event.params._additionalRepositoryIndex;
+  additionalRepository.save();
+}
+
+export function handleDependencyAdditionalRepositoryRemoved(
+  event: DependencyAdditionalRepositoryRemoved
+): void {
+  const dependency = Dependency.load(event.params._dependencyType.toString());
+  if (!dependency) {
+    return;
+  }
+  const id = generateDependencyAdditionalRepositoryId(
+    dependency.id,
+    event.params._additionalRepositoryIndex
+  );
+
+  const additionalRepository = DependencyAdditionalRepository.load(id);
+  if (!additionalRepository) {
+    return;
+  }
+  store.remove("DependencyAdditionalRepository", additionalRepository.id);
+
+  // Load each additional repository after the removed one and decrement their index
+  for (
+    let i = event.params._additionalRepositoryIndex.toI32() + 1;
+    i < dependency.additionalRepositoryCount.toI32();
+    i++
+  ) {
```

Comment:
I think this has to be updated in the same way as `handleDependencyAdditionalCDNRemoved`